### DISCUSSION TITLE
A simple question about synthetic graphs

### DISCUSSION THREAD

**Discussion by harpomaxx** [danielegrattarola/spektral#430] [General]
I'm trying to do transductive node prediction. I'm testing a GCN model with synthetic graphs generated using the following code:

```python
# Function to generate a synthetic graph
def generate_synthetic_graph(n_nodes, n_features, n_classes):
    # Create features matrix filled with ones
    features = np.ones((n_nodes, n_features))
    # Generate a random adjacency matrix
    adj_matrix = np.random.randint(0, 2, size=(n_nodes, n_nodes))
    adj_matrix = np.tril(adj_matrix) + np.tril(adj_matrix, -1).T
    # Create the identity matrix and add it to the adjacency matrix
    identity_matrix = np.eye(n_nodes)
    adj_matrix = adj_matrix + identity_matrix
    # Normalize the adjacency matrix
    adj_matrix = normalized_adjacency(adj_matrix)
    # Convert to SparseTensor
    adj_matrix = sparse.csr_matrix(adj_matrix)
    # Generate random labels and convert them to one-hot encoding
    labels = np.random.randint(0, n_classes, n_nodes)
    labels = tf.keras.utils.to_categorical(labels)
    # Return a Graph object with features, adjacency matrix, and labels
    return Graph(x=features, a=adj_matrix, y=labels)
```
Then I subclass the `Dataset` class to incorporate two datasets: one for training and the other for testing.

```python
# Custom dataset class for synthetic graphs
class SyntheticDataset(Dataset):
    def __init__(self, graphs=None, **kwargs):
        self.graphs = graphs
        super().__init__(**kwargs)

    # Read the graphs from the dataset
    def read(self):
        return self.graphs
        
# Generate synthetic graphs
n_graphs = 2
n_nodes = 100
n_features = 10
n_classes = 2
graphs = [generate_synthetic_graph(n_nodes, n_features, n_classes) for _ in range(n_graphs)]
# Split the graphs into training and testing sets
train_graphs, test_graphs = train_test_split(graphs, test_size=0.5, random_state=42)
# Create training and testing datasets
train_dataset = SyntheticDataset(train_graphs)
test_dataset = SyntheticDataset(test_graphs)
```

I'm training a GCN() model and after some epochs my training loss get stuck to 0.6 and the accuracy to 0.5. 

In a sense this is excepted since nodes were labeled randomly. But at least it  shouldn't be possible to overfit the training ? I mean given a complex enough model.. I shouldn't have better result than a 0.5 accuracy? 

I've have removed `dropout` and `l2_reg` values.

I hope my question is clear enough...

**Comment by danielegrattarola** [comment]
Hi, 

fitting random labels is possible but I don't think it's necessarily straightforward. 
How many parameters/layers does the model have?
